src/util/simplify_expr.cpp

```diff
@@ -1870,7 +1872,7 @@ simplify_exprt::simplify_byte_extract(const byte_extract_exprt &expr)
     expr2bits(expr.op(), expr.id() == ID_byte_extract_little_endian, ns);
 
   if(
-    bits.has_value() &&
+    offset.has_value() && bits.has_value() &&
```

Comment:
What is this doing exactly ? seems like it restricts the optimization to a subset of cases ?